### ISSUE TITLE
control tree mixin needs updating

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1024]
In GitLab by @jvande42b on May 8, 2018, 04:08

In at least one place, control tree mixin is using casadi24 syntax, which results in runtime errors:
https://gitlab.com/deltares/rtc-tools/blob/master/src/rtctools/optimization/control_tree_mixin.py#L39

This mixin needs to be fixed and tested. @vreeken I don't know why the tests are passing here...

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 8, 2018, 11:04

I think @roalva82 had a case that failed, and I fixed this locally on his PC. 

The reason the test case does not fail is because it overrides the default `control_tree_options` method which contains the mistake.

```python
    def control_tree_options(self):
        return {'forecast_variables': ['constant_input'],
                'branching_times': self._branching_times,
                'k': 2}
```

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 11, 2018, 14:57

Would it make sense to write tests so that when we overwrite a method, we always call super()? That way our tests don't mask problems.

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 11, 2018, 16:14

mentioned in commit 1ac2f9c2c67b7ad7f3ccf18a204518187cedf968

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 11, 2018, 16:49

mentioned in commit e364a63b8057a2d7910c899d991019a5786b9c67

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on May 14, 2018, 24:25

Yes, that very much makes sense. We should also take a look at the coverage report, to see which lines are covered by tests in the Mixin files and which are not. Especially tests for methods like this, that we expect to be executed with a super() call but are actually not, should be fixed.

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 14, 2018, 17:28

Cool. new issue discussing tests: #1025

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 15, 2018, 19:34

mentioned in commit 3088a98fce8f8bb5c8986f1613ce9617d291f46c

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 15, 2018, 19:40

closed via merge request !110

**Comment by SGeeversAtVortech** [comment]
In GitLab by @jvande42b on May 15, 2018, 19:40

closed via commit 3088a98fce8f8bb5c8986f1613ce9617d291f46c
